scripts/import-reborn-run-artifact.py

```
@@ -121,36 +121,93 @@ def trace_candidate(artifact: dict[str, Any], model_override: str | None) -> dic
+        # silently merging unscoped records into a neighboring run.
+        key = run_id or f"unscoped:{message.get('sequence', len(grouped))}"
+        grouped.setdefault(key, []).append(message)
+    return list(grouped.values()), skipped_unscoped
+
+
+def trace_candidate(artifact: dict[str, Any], model_override: str | None) -> dict[str, Any]:
+    turns: list[dict[str, Any]] = []
+    captured_models: list[str] = []
+    message_groups, skipped_unscoped = artifact_turns(artifact)
+    if not message_groups:
+        raise ValueError("thread artifact has no complete run-scoped replayable turns")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)